src/bin/jcli_app/utils/io.rs

```diff
@@ -1,37 +1,37 @@
+use std::io::{stdin, stdout, BufRead, BufReader, Error, Write};
 use std::path::Path;
 
 /// open the given file path as a writable stream, or stdout if no path
 /// provided
-pub fn open_file_write<P: AsRef<Path>>(path: &Option<P>) -> Box<dyn std::io::Write> {
-    if let Some(path) = path {
-        Box::new(
-            std::fs::OpenOptions::new()
+pub fn open_file_write<P: AsRef<Path>>(path: &Option<P>) -> Result<impl Write, Error> {
+    match path {
+        Some(path) => {
+            let writer = std::fs::OpenOptions::new()
                 .create(true)
                 .write(true)
                 .read(false)
                 .append(false)
-                .open(path)
-                .unwrap(),
-        )
-    } else {
-        Box::new(std::io::stdout())
+                .truncate(true)
+                .open(path)?;
```

Comment:
Good. You may want to check that the I/O error message captures the information so no `map_err` is needed to augment it with the file path.

Reply:
I've checked it previously and no, it returns error which prints vague message that file does not exist. The error also has no source. Correct me if I'm wrong, but I think it could be slightly dangerous to depend on undocumented formatting of the message anyway, even if it's nothing serious.

Reply:
Yes, that's why there could be a `map_err` which will reuse the original error's code, but prepend the file path information to the message of the returned error.

Reply:
That would make sense. Unfortunately the path is not captured, but that's not a problem, we'll just add path in higher-level errors.

Reply:
That's a solution to, no real need to encode it in errors here.